### ISSUE TITLE
Django REST Framework integration

### ISSUE THREAD

**Issue by barseghyanartur** [barseghyanartur/ska#3]
Django REST Framework integration for password-less authentication.

*WIP in branch [restframework-integration](https://github.com/barseghyanartur/ska/tree/restframework-integration)*

**Permissions classes**

For protecting views without actually being authenticated into the system, the following permission classes shall be implemented. They shall work for plan settings and provider settings, as well as both plain- and provider-settings shall work in combination with `django-constance` as well.

- [x] ``SignedRequestRequired``
- [x] ``ProviderSignedRequestRequired``
- [x] ``ConstanceSignedRequestRequired``
- [x] ``ConstanceProviderSignedRequestRequired``

**Obtaining of token**

Login view for obtaining the authentication token. Same requirements: plan, provider + both of them to work with `django-constance`.

**Comment by barseghyanartur** [comment]
**Example of ``ProviderSignedRequestRequired`` permission class usage**

```python
from rest_framework.viewsets import ModelViewSet

from ska.contrib.django.ska.integration.drf.permissions import (
    ProviderSignedRequestRequired
)

from .models import FooItem
from .serializers import FooItemSerializer

__all__ = (
    'FooItemViewSet',
)


class FooItemViewSet(ModelViewSet):
    """FooItem model viewset."""

    permission_classes = (ProviderSignedRequestRequired,)
    queryset = FooItem.objects.all()
    serializer_class = FooItemSerializer
```

**Sample endpoint URLs**

- [Access denied URL](http://localhost:8000/foo/api/fooitem/)
- [Access granted URL](http://localhost:8000/foo/api/fooitem/?signature=JxQAf3vQeLJ6VMKzld1aRNHl8yg%3D&auth_user=test_ska_user_client_1.admins&valid_until=1545871479.0&extra=email%2Cfirst_name%2Clast_name%2Cprovider&email=test_ska_user_client_1.admins%40mail.example.com&first_name=John+client_1.admins&last_name=Doe+client_1.admins&provider=client_1.admins)

**See the example project**

Check [DRF example](http://localhost:8000/foo/drf/) of the example project.

**Comment by barseghyanartur** [comment]
Implemented in 1.7.
